server/sync/worker.go

```diff
@@ -24,21 +25,23 @@ import (
 
 // Worker processes sync work items.
 type Worker struct {
-	id        int
-	db        types.SyncDatabaseAPI
-	store     types.StoreAPI
-	workQueue <-chan synctypes.WorkItem
-	timeout   time.Duration
+	id             int
+	db             types.DatabaseAPI
+	store          types.StoreAPI
+	workQueue      <-chan synctypes.WorkItem
+	timeout        time.Duration
+	monitorService *monitor.MonitorService
 }
 
 // NewWorker creates a new worker instance.
-func NewWorker(id int, db types.SyncDatabaseAPI, store types.StoreAPI, workQueue <-chan synctypes.WorkItem, timeout time.Duration) *Worker {
+func NewWorker(id int, db types.DatabaseAPI, store types.StoreAPI, workQueue <-chan synctypes.WorkItem, timeout time.Duration, monitorService *monitor.MonitorService) *Worker {
```

Comment:
there is a hard guarantee assumption here that, if there are no syncs running, we did the job of indexing them. i would really make the monitoring service run even if there are no syncs running (decouple it from sync). consider the situation where:
- there is a running sync
- we indexed 50% of the sync data
- sync is completed (ie. the data was sync)
- the indexing/monitoring exists because there is no sync anymore

NOTE: This assumes that we have a way to know the state of a sync (running/completed/etc), but even without this scenario, the same can happen by, instead of having a state for sync completion, I decide to delete the sync.

Reply:
good point. we can either we make sure we run a full post-sync operation after the sync is deleted or we need to decouple